src/configobj/__init__.py

```diff
@@ -757,7 +758,7 @@ def merge(self, indict, decoupled=False):
         """
         for key, val in list(indict.items()):
             if decoupled:
-                val = copy.deepcopy(val)
+                val = copy(val)
```

Comment:
Just so I understand you: Take all the regexp into a utile class (with the delimiter parameter to `__init__()` and have ConfigObj instantiate one of those?

Also, I am unsure what you mean by all of them becoming dynamic. As far as I can see, only one (`_keyword`) uses the delimiter. Can you provide a test for what you mean?

Thinking about it, maybe '#' should not be allowed as a delimiter… Is there any other character that need to be excluded?